src/Model/Resolver/Quote/Data/SalesRuleLabel.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Rapidez\Compadre\Model\Resolver\Quote\Data;
+
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\SalesRule\Api\RuleRepositoryInterface;
+use Magento\SalesRule\Api\Data\RuleInterface;
+use Magento\Quote\Model\Quote\Item;
+
+class SalesRuleLabel implements ResolverInterface
+{
+    public function __construct(
+        protected RuleRepositoryInterface $ruleRepository
+    )
+    {}
+
+    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null)
+    {
+        /** @var Item */
+        $cartItem = $value['model'];
+
+        $labels = [];
+
+        foreach (explode(',', (string) $cartItem->getAppliedRuleIds()) as $ruleId) {
+            /** @var RuleInterface $rule */
+            $rule = $this->ruleRepository->getById((int) $ruleId);
+
+            $labels[] = $rule->getName();
```

Comment:
I think we might want to expand this to a Graphql salesRuleInterface (see: https://github.com/magento/magento2/blob/ea3216321185052523ad65728ee97a53fb8bdb86/app/code/Magento/GiftMessageGraphQl/etc/schema.graphqls#L41), so we can return more than just the name.

I'd expect we'd rather want to show the `storeLabel` instead of the name, since the name is for internal use without localization.
As well as the SimpleAction, and DiscountAmount so the customer could be shown the discount for just that item

Reply:
I think i added all the relevant data.